### PR TITLE
Tests: Add checksum validation to the postgen datapackage test

### PR DESCRIPTION
**Pull request by alwaysintreble**
## What is this fixing or adding?
title. I think this wasn't originally added because there are plans to make the datapackage (and so the checksum) dynamic, but this test should still be valid after the API changes since it just verifies that the datapackage getting packed matches the checksum that goes with it.

### CODE REVIEW

File: test/general/test_ids.py [hunk 1]
```diff
@@ -57,6 +58,9 @@ def test_duplicate_location_ids(self):
     def test_postgen_datapackage(self):
         """Generates a solo multiworld and checks that the datapackage is still valid"""
         for gamename, world_type in AutoWorldRegister.world_types.items():
+            # because of test/general/__init__.py:56
+            if world_type.hidden:
+                continue
```
Comment (black-sliver) [inline test/general/test_ids.py:63]: I think that check should probably move to the relevant code piece and maybe be different - such as `if gamename in ...`

docstring of hidden says "... Does not remove functionality.".

**Review by black-sliver — COMMENTED** [review]
> but this test should still be valid after the API changes

Once we allow worlds modifying data package during gen, we'll need to move away from the global `network_data_package`, so this test will break then, but that's fine.

But see the comment below.

**Comment by alwaysintreble** [comment]
unsure if this is the best fix. alternative is to add the test world to `network_data_package` when it gets loaded in, which should also work and would still be relevant after we have full lazy loading.

**Comment by black-sliver** [comment]
Oh, i meant more like `if gamename in network_data_package["games"]:` then do the check

**Comment by alwaysintreble** [comment]
> Oh, i meant more like `if gamename in network_data_package["games"]:` then do the check

this sounds like a future bug to be introduced. took the route of just adding the test game to the data package. will likely need to add stuff to its lookups in the future for other tests (such as item and location related options), so testing its datapackage is valid offers a bit of value

**Review by black-sliver — APPROVED** [review]
Not sure where this will move, but approving for now in hopes that this may find worlds that modify name_to_id. I think if someone git blames a test failure, it should be easy to work out why this is here and what to change in the future.

**Pull request merged**